enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/PredefinedRolesBuilder.java

```diff
@@ -28,14 +28,15 @@
 import java.util.Collections;
 import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.Set;
 
 import static org.neo4j.server.security.enterprise.auth.plugin.api.PredefinedRoles.*;
 
 public class PredefinedRolesBuilder implements RolesBuilder
 {
     public static final Map<String,SimpleRole> roles = staticBuildRoles();
 
-    public static Map<String,SimpleRole> staticBuildRoles()
+    private static Map<String,SimpleRole> staticBuildRoles()
     {
         Map<String, SimpleRole> roles = new LinkedHashMap<>( 4 );
```

Comment:
Why `LinkedHashMap`? Why does this method put a _mutable_ map in a `public static final` field of a `public` class? I would recommend wrapping the returned map in `Collections.unmodifiableMap`, and have the inner map be a `ConcurrentHashMap` because `setAllowPublisherTokenCreate` can modify it at runtime.

Also, how does this being a static resource interact when multiple databases, with different configurations, are started up in the same Java process?